extensions/emmet/src/abbreviationActions.ts

```diff
@@ -59,55 +47,95 @@ function doWrapping(_: boolean, args: any) {
 	const syntax = getSyntaxFromArgs(args) || 'html';
 	const rootNode = parseDocument(document, true);
 
-	let inPreview = false;
-	let currentValue = '';
 	const helper = getEmmetHelper();
 
-	// Fetch general information for the succesive expansions. i.e. the ranges to replace and its contents
-	const rangesToReplace: PreviewRangesWithContent[] = editor.selections.sort((a: vscode.Selection, b: vscode.Selection) => { return a.start.compareTo(b.start); }).map(selection => {
-		let rangeToReplace: vscode.Range = selection.isReversed ? new vscode.Range(selection.active, selection.anchor) : selection;
+	const operationRanges = editor.selections.sort((a, b) => a.start.compareTo(b.start)).map(selection => {
+		let rangeToReplace: vscode.Range = selection;
+		// wrap around the node if the selection falls inside its open or close tag
+		{
+			let { start, end } = rangeToReplace;
+
+			const startOffset = document.offsetAt(start);
+			const startNode = getFlatNode(rootNode, startOffset, true) as (HtmlNode | undefined);
+			if (
```

Comment:
This if statement seems pretty big; I'm wondering whether the conditional (not the body) can be moved out to its own function? That way, one doesn't need to repeat the whole conditional for `endNode` later on.